lib/handle_callback.ts

```diff
@@ -19,12 +18,44 @@ import {
 import { getAndDeleteOAuthSession, setSiteSession } from "./_kv.ts";
 
 /** Options for {@linkcode handleCallback}. */
-export interface HandleCallbackOptions {
-  /** Overwrites cookie properties set in the response. These must match the
+export interface HandleCallbackOptions<T extends unknown = unknown> {
+  /**
+   * Overwrites cookie properties set in the response. These must match the
    * cookie properties used in {@linkcode getSessionId} and
    * {@linkcode signOut}.
    */
   cookieOptions?: Partial<Cookie>;
+  /**
+   * Function that uses the access token to get the session object. This is
+   * useful for fetching the user's profile from the OAuth provider. If
+   * undefined, the session object will be set to `undefined`.
```

Comment:
Is it useful to allow `undefined` for this property?

Reply:
I thought it might be worth doing for a use case when the dev doesn't care about or store any session data. Is that a reasonable use case? WDYT?

Reply:
If it not a likely/reasonable use case (which could be the case), I'll make this the 3rd argument of the function.

Reply:
Is there any way to identify the user without session data with current set of APIs? I guess there's none (because `getSessionId` is removed), and if there's no way to identify the user, I think the app can't do anything interesting about the user..